src/test/java/kg/jarkyn/cobspecserver/ServerTest.java

```diff
@@ -1,43 +1,45 @@
 package kg.jarkyn.cobspecserver;
 
 import kg.jarkyn.cobspecserver.doubles.ClientDouble;
+import kg.jarkyn.cobspecserver.doubles.ExecutorServiceDouble;
 import kg.jarkyn.cobspecserver.doubles.ListenerDouble;
 import kg.jarkyn.cobspecserver.doubles.ResponseControllerDouble;
 import org.junit.Before;
 import org.junit.Test;
 
 import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
 
 public class ServerTest {
 
     private ClientDouble client;
     private ResponseControllerDouble controller;
+    private ExecutorServiceDouble executor;
 
     @Before
     public void setUp() throws Exception {
         client = new ClientDouble();
+        executor = new ExecutorServiceDouble();
+        controller = new ResponseControllerDouble();
     }
 
     @Test
     public void satisfiesReceivedRequest() {
-        Server server = new Server(new ListenerDouble(client), controllerWithResponse("hello"));
+        Server server = new Server(executor, new ListenerDouble(client), controller);
 
         server.satisfyRequest();
 
-        assertEquals("hello", client.getReceivedMessage());
+        assertTrue(executor.hasExecuted());
+        assertTrue(executor.getRunnable() instanceof Worker);
     }
 
     @Test
     public void runsUntilStopped() {
```

Comment:
I first thought: _when is the server stopped?_. Then I looked into `ListenerDouble` because that magic `2` seemed to be related.
It is not clear that `ListenerDouble` stops after two connections. 

But I wonder: Is `ListenerSocket` the appropriate place to stop the server? When should the server stop? why?
The only way to make `ListenerSocket` stop is by calling its `stop()` method. How should one stop the server?
```
ListenerSocket listener = createListener();
Server server = new Server(listener, controller);
server.start();
listener.stop();
```
Seems like the server is not encapsulating its internals: we still need to know its content to be able to do things with it.
What about making the `Server` able to stop?
```
Server server = new Server(createListener(), controller);
server.start();
server.stop();
```

Don't know the value of this though...